manifests/global_hook.pp

```diff
@@ -52,22 +52,19 @@
 #
 # Copyright 2017 Hidde Boomsma
 #
-define gitlab::global_hook(
-  $type = undef,
-  $content = undef,
-  $source = undef,
-  $custom_hooks_dir = undef
+define gitlab::global_hook (
+  Pattern[/^(post-receive|pre-receive|update)$/] $type,
```

Comment:
`Enum` is a better choice here?

Reply:
`Enum['post-receive', 'pre-receive', 'update'] $type`

Reply:
Sure :)